morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -320,17 +320,17 @@ void process(final MappedClass mc, final Validation validation) {
             LOG.trace("Executing findAndModify(" + dbColl.getName() + ") with delete ...");
         }
 
-        FindAndModifyOptions copy = options.copy()
-                                           .projection(query.getFieldsObject())
-                                           .sort(query.getSortObject())
-                                           .returnNew(false)
-                                           .upsert(false)
-                                           .remove(true);
+        FindAndModifyOptions copy = enforceWriteConcern(options, query)
```

Comment:
Can you add a regression test to ensure that `enforceWriteConcern` works as expected.